game/scripts/vscripts/items/drums_of_endurance.lua

```diff
@@ -0,0 +1,259 @@
+LinkLuaModifier( "modifier_item_drums_of_endurance_oaa_active", "items/drums_of_endurance.lua", LUA_MODIFIER_MOTION_NONE )
+
+item_drums_of_endurance_oaa = class(ItemBaseClass)
+
+function item_drums_of_endurance_oaa:GetIntrinsicModifierName()
+  return "modifier_intrinsic_multiplexer"
+end
+
+function item_drums_of_endurance_oaa:GetIntrinsicModifierNames()
+  return {
+    "modifier_item_drums_of_endurance_oaa",
+    "modifier_item_drums_of_endurance_oaa_swiftness_aura",
+    "modifier_item_drums_of_endurance_oaa_swiftness_aura_effect"
+  }
+end
+
+------------------------------------------------------------------------------------------------------------------------------
+--On Casting/Activating Item
+
+function item_drums_of_endurance_oaa:OnSpellStart()
+  --Initializing needed variables
+  local ability = self
+  local caster = ability:GetCaster()
+  local casterTeam = caster:GetTeamNumber()
+
+  local units = FindUnitsInRadius(
+    casterTeam,
+    caster:GetAbsOrigin(),
+    nil,
+    ability:GetSpecialValueFor("radius"),
+    DOTA_UNIT_TARGET_TEAM_FRIENDLY,
+    bit.bor(DOTA_UNIT_TARGET_BASIC, DOTA_UNIT_TARGET_HERO),
+    DOTA_UNIT_TARGET_FLAG_INVULNERABLE,
+    FIND_ANY_ORDER,
+    false
+  )
+
+  local function EnduranceActive(unit)
+    unit:AddNewModifier(self:GetCaster(), self, "modifier_item_drums_of_endurance_oaa_active", {duration = self:GetSpecialValueFor("duration")})
+  end
+  --Applying_Active_Effect_to_allied_units
+  units = iter(units)
+  foreach(EnduranceActive,units)
+
+end
+
+------------------------------------------------------------------------------------------------------------------------------
+--Active_Modifier
+
+modifier_item_drums_of_endurance_oaa_active = class(ModifierBaseClass)
+
+function modifier_item_drums_of_endurance_oaa_active:OnCreated()
+  --Ability_specials
+  self.active_attack_speed = self:GetAbility():GetSpecialValueFor("bonus_attack_speed_pct")
+  self.active_movement_speed = self:GetAbility():GetSpecialValueFor("bonus_movement_speed_pct")
+end
+
+function modifier_item_drums_of_endurance_oaa_active:DeclareFunctions()
+  local decFuncs = {
+    MODIFIER_PROPERTY_MOVESPEED_BONUS_CONSTANT,
+    MODIFIER_PROPERTY_ATTACKSPEED_BONUS_CONSTANT
+  }
+  return decFuncs
+end
+
+function modifier_item_drums_of_endurance_oaa_active:GetModifierMoveSpeedBonus_Constant()
+  return self.active_movement_speed
+end
+
+function modifier_item_drums_of_endurance_oaa_active:GetModifierAttackSpeedBonus_Constant()
+  return self.active_attack_speed
+end
+
+------------------------------------------------------------------------------------------------------------------------------
+--Aura
+
+modifier_item_drums_of_endurance_oaa_swiftness_aura = class(ModifierBaseClass)
+
+function modifier_item_drums_of_endurance_oaa_swiftness_aura:OnCreated()
+  self.radius = self:GetAbility():GetSpecialValueFor("radius")
+end
```

Comment:
For future reference, it's preferable to put the aura functionality on the intrinsic modifier instead of a separate one if it's just an always on aura.

Reply:
Thanks, will keep that in mind